File: RxDemo/src/main/java/com/vondear/rxdemo/activity/ActivityRxCaptcha.java
package com.vondear.rxdemo.activity;

import android.graphics.drawable.Drawable;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.Button;
import android.widget.ImageView;
import android.widget.LinearLayout;
import android.widget.SeekBar;
import android.widget.TextView;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.bumptech.glide.Glide;
import com.bumptech.glide.request.target.SimpleTarget;
import com.bumptech.glide.request.transition.Transition;
import com.vondear.rxdemo.R;
import com.vondear.rxtool.RxDeviceTool;
import com.vondear.rxtool.view.RxToast;
import com.vondear.rxui.activity.ActivityBase;
import com.vondear.rxui.view.RxCaptcha;
import com.vondear.rxui.view.RxTitle;
import com.vondear.rxui.view.swipecaptcha.RxSwipeCaptcha;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.OnClick;

import static com.vondear.rxui.view.RxCaptcha.TYPE.CHARS;


/**
 * @author vondear
 */
public class ActivityRxCaptcha extends ActivityBase {

    @BindView(R.id.tv_code)
    TextView tvCode;
    @BindView(R.id.iv_code)
    ImageView ivCode;
    @BindView(R.id.btn_get_code)
    Button btnGetCode;
    @BindView(R.id.swipeCaptchaView)
    RxSwipeCaptcha mRxSwipeCaptcha;
    @BindView(R.id.dragBar)
    SeekBar mSeekBar;
    @BindView(R.id.btnChange)
    Button mBtnChange;
    @BindView(R.id.activity_rx_captcha)
    LinearLayout mActivityRxCaptcha;
    @BindView(R.id.rx_title)
    RxTitle mRxTitle;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_rx_captcha);
        ButterKnife.bind(this);
        RxDeviceTool.setPortrait(this);
        initView();
    }

    private void initView() {
        mRxTitle.setLeftFinish(mContext);
        mRxSwipeCaptcha.setOnCaptchaMatchCallback(new RxSwipeCaptcha.OnCaptchaMatchCallback() {
            @Override
            public void matchSuccess(RxSwipeCaptcha rxSwipeCaptcha) {
                RxToast.success(mContext, "验证通过！", Toast.LENGTH_SHORT).show();
                //swipeCaptcha.createCaptcha();
                mSeekBar.setEnabled(false);
            }

            @Override
            public void matchFailed(RxSwipeCaptcha rxSwipeCaptcha) {
                Log.d("zxt", "matchFailed() called with: rxSwipeCaptcha = [" + rxSwipeCaptcha + "]");
                RxToast.error(mContext, "验证失败:拖动滑块将悬浮头像正确拼合", Toast.LENGTH_SHORT).show();
                rxSwipeCaptcha.resetCaptcha();
                mSeekBar.setProgress(0);
            }
        });
        mSeekBar.setOnSeekBarChangeListener(new SeekBar.OnSeekBarChangeListener() {

            @Override
            public void onProgressChanged(SeekBar seekBar, int progress, boolean fromUser) {
                mRxSwipeCaptcha.setCurrentSwipeValue(progress);
            }

            @Override
            public void onStartTrackingTouch(SeekBar seekBar) {
                //随便放这里是因为控件
                mSeekBar.setMax(mRxSwipeCaptcha.getMaxSwipeValue());
            }

            @Override
            public void onStopTrackingTouch(SeekBar seekBar) {
                Log.d("zxt", "onStopTrackingTouch() called with: seekBar = [" + seekBar + "]");
                mRxSwipeCaptcha.matchCaptcha();
            }
        });

        SimpleTarget<Drawable> simpleTarget = new SimpleTarget<Drawable>() {
            @Override
            public void onResourceReady(@NonNull Drawable resource, @Nullable Transition<? super Drawable> transition) {
                mRxSwipeCaptcha.setImageDrawable(resource);
                mRxSwipeCaptcha.createCaptcha();
            }
        };

        //测试从网络加载图片是否ok
        Glide.with(this)
                .load(R.drawable.douyu)
                .into(simpleTarget);
    }

    @OnClick({R.id.btn_get_code, R.id.btnChange})
    public void onClick(View view) {
        switch (view.getId()) {
            case R.id.btn_get_code:

                RxCaptcha.build()
                        .backColor(0xffffff)
                        .codeLength(6)
                        .fontSize(60)
                        .lineNumber(0)
                        .size(200, 70)
                        .type(CHARS)
                        .into(ivCode);

                tvCode.setText(RxCaptcha.build().getCode());
                break;
            case R.id.btnChange:
                mRxSwipeCaptcha.createCaptcha();
                mSeekBar.setEnabled(true);
                mSeekBar.setProgress(0);
                break;
            default:
                break;
        }
    }
}

File: RxDemo/src/main/java/com/tamsiree/rxdemo/activity/ActivityRxCaptcha.kt
package com.tamsiree.rxdemo.activity

import android.graphics.drawable.Drawable
import android.os.Bundle
import android.widget.SeekBar
import android.widget.SeekBar.OnSeekBarChangeListener
import android.widget.Toast
import com.bumptech.glide.Glide
import com.bumptech.glide.request.target.SimpleTarget
import com.bumptech.glide.request.transition.Transition
import com.tamsiree.rxdemo.R
import com.tamsiree.rxkit.RxDeviceTool.setPortrait
import com.tamsiree.rxkit.TLog
import com.tamsiree.rxkit.view.RxToast
import com.tamsiree.rxui.activity.ActivityBase
import com.tamsiree.rxui.view.RxCaptcha
import com.tamsiree.rxui.view.swipecaptcha.RxSwipeCaptcha
import com.tamsiree.rxui.view.swipecaptcha.RxSwipeCaptcha.OnCaptchaMatchCallback
import kotlinx.android.synthetic.main.activity_rx_captcha.*

/**
 * @author tamsiree
 */
class ActivityRxCaptcha : ActivityBase() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_rx_captcha)
        setPortrait(this)
        initView()
    }

    override fun initView() {
        rx_title.setLeftFinish(mContext)
        swipeCaptchaView.onCaptchaMatchCallback = object : OnCaptchaMatchCallback {
            override fun matchSuccess(rxSwipeCaptcha: RxSwipeCaptcha) {
                RxToast.success(mContext, "验证通过！", Toast.LENGTH_SHORT)?.show()
                //swipeCaptcha.createCaptcha();
                dragBar.isEnabled = false
            }

            override fun matchFailed(rxSwipeCaptcha: RxSwipeCaptcha) {
                TLog.d("zxt", "matchFailed() called with: rxSwipeCaptcha = [$rxSwipeCaptcha]")
                RxToast.error(mContext, "验证失败:拖动滑块将悬浮头像正确拼合", Toast.LENGTH_SHORT)?.show()
                rxSwipeCaptcha.resetCaptcha()
                dragBar.progress = 0
            }
        }
        dragBar.setOnSeekBarChangeListener(object : OnSeekBarChangeListener {
            override fun onProgressChanged(seekBar: SeekBar, progress: Int, fromUser: Boolean) {
                swipeCaptchaView.setCurrentSwipeValue(progress)
            }

            override fun onStartTrackingTouch(seekBar: SeekBar) {
                //随便放这里是因为控件
                dragBar.max = swipeCaptchaView.maxSwipeValue
            }

            override fun onStopTrackingTouch(seekBar: SeekBar) {
                TLog.d("zxt", "onStopTrackingTouch() called with: seekBar = [$seekBar]")
                swipeCaptchaView.matchCaptcha()
            }
        })
        val simpleTarget: SimpleTarget<Drawable?> = object : SimpleTarget<Drawable?>() {
            override fun onResourceReady(resource: Drawable, transition: Transition<in Drawable?>?) {
                swipeCaptchaView.setImageDrawable(resource)
                swipeCaptchaView.createCaptcha()
            }
        }

        //测试从网络加载图片是否ok
        Glide.with(this)
                .load(R.drawable.douyu)
                .into(simpleTarget)

        btn_get_code.setOnClickListener {
            RxCaptcha.build()
                    .backColor(0xffffff)
                    .codeLength(6)
                    .fontSize(60)
                    .lineNumber(0)
                    .size(200, 70)
                    .type(RxCaptcha.TYPE.CHARS)
                    .into(iv_code)
            tv_code.text = RxCaptcha.build().code
        }
        btnChange.setOnClickListener {
            swipeCaptchaView.createCaptcha()
            dragBar.isEnabled = true
            dragBar.progress = 0
        }
    }

    override fun initData() {

    }

}